src/main/scala/model/Bunny.scala

```diff
@@ -0,0 +1,142 @@
+package model
+
+import scala.util.Random
+
+/**
+ * Represents a Bunny.
+ */
+sealed trait Bunny{
+  val genotype: CompletedGenotype
+  val mom: Option[Bunny]
+  val dad: Option[Bunny]
+  val age: Int
+  val isAlive: Boolean
+
+  override def toString: String = {
+    super.toString + "\n" + genotype.genes
+      .map(g => "\t" + g._1 + ": "+ g._2.getVisibleTrait.toString.toLowerCase + " (" + g._2.getLetters + ")")
+      .reduce(_ + "\n" + _)
+      .replace("_", " ") + "\n"
+  }
+}
+
+/**
+ * Represents a Bunny which is still alive.
+ * @param genotype  contains the Bunny genes, it must be completed (contain all the available Genes of the world)
+ * @param mom       the bunny's mom
+ * @param dad       the bunny's dad
+ * @param age       how many generations the Bunny has spent in the world.
+ */
+case class AliveBunny(genotype: CompletedGenotype, mom:Option[Bunny], dad:Option[Bunny], age:Int) extends Bunny{
+  override val isAlive: Boolean = true
+}
+
+/**
+ * Represents a Bunny which dead.
+ * @param genotype  contains the Bunny genes, it must be completed (contain all the available Genes of the world)
+ * @param mom       the bunny's mom
+ * @param dad       the bunny's dad
+ * @param age       how many generations the Bunny has spent in the world.
+ */
+case class DeadBunny(genotype: CompletedGenotype, mom:Option[Bunny], dad:Option[Bunny], age:Int) extends Bunny{
+  override val isAlive: Boolean = false
+}
+
+/**
+ * Represents the abstraction of a Bunny that as just been created.
+ */
+abstract class InitBunny() extends Bunny{
+  override val age: Int = 0
+  override val isAlive: Boolean = true
+}
+
+/**
+ * Represents a Bunny that as just been created.
+ */
+case class ChildBunny(genotype: CompletedGenotype, mom:Option[Bunny], dad:Option[Bunny]) extends InitBunny
+
+/**
+ * Represents the first Bunny which appears in the world, so it does not have a mom and a dad.
+ */
+case class FirstBunny(genotype: CompletedGenotype) extends InitBunny {
+  override val mom: Option[Bunny] = Option.empty
+  override val dad: Option[Bunny] = Option.empty
+}
+
+object BunnyUtils {
```

Comment:
Secondo me questo object può essere spostato in un file diverso ed essere chiamato tipo Reproduction perchè mi sembra contenere solo metodi riguardanti la riproduzione dei conigli

Reply:
Si puó fare anche così, l'avevo lasciato insieme al Bunny perchè l'avevo visto come un suo "Companion Object" con metodi statici che lo riguardano.